presto-parquet/src/main/java/com/facebook/presto/parquet/cache/MetadataReader.java

```diff
@@ -127,34 +184,61 @@ public static ParquetFileMetadata readFooter(ParquetDataSource parquetDataSource
+                    boolean encryptedMetadata = false;
+
+                    if (null == cryptoMetaData) { // Plaintext column
+                        columnPath = getPath(metaData);
+                        if (fileDecryptor.isPresent() && !fileDecryptor.get().plaintextFile()) {
+                            // mark this column as plaintext in encrypted file decryptor
+                            fileDecryptor.get().setColumnCryptoMetadata(columnPath, false, false, (byte[]) null, columnOrdinal);
+                        }
+                    }
+                    else {  // Encrypted column
+                        if (cryptoMetaData.isSetENCRYPTION_WITH_FOOTER_KEY()) { // Column encrypted with footer key
+                            if (!encryptedFooter) {
+                                throw new ParquetCryptoRuntimeException("Column encrypted with footer key in file with plaintext footer");
+                            }
+                            if (null == metaData) {
+                                throw new ParquetCryptoRuntimeException("ColumnMetaData not set in Encryption with Footer key");
+                            }
+                            if (!fileDecryptor.isPresent()) {
+                                throw new ParquetCryptoRuntimeException("Column encrypted with footer key: No keys available");
+                            }
+                            columnPath = getPath(metaData);
+                            fileDecryptor.get().setColumnCryptoMetadata(columnPath, true, true, (byte[]) null, columnOrdinal);
+                        }
+                        else { // Column encrypted with column key
+                            encryptedMetadata = true;
+                        }
+                    }
+
+                    try {
+                        if (encryptedMetadata) {
+                            // TODO: We decrypted data before filter projection. This could send unnecessary traffic to KMS. This so far not seen a problem in production.
+                            //  In parquet-mr, it uses lazy decryption but that required to change ColumnChunkMetadata. We will improve it alter.
+                            metaData = decryptMetadata(rowGroup, cryptoMetaData, columnChunk, fileDecryptor.get(), columnOrdinal);
+                            columnPath = getPath(metaData);
+                        }
+                        ColumnChunkMetaData column = buildColumnChunkMetaData(metaData, columnPath, messageType.getType(columnPath.toArray()).asPrimitiveType());
+                        column.setColumnIndexReference(toColumnIndexReference(columnChunk));
+                        column.setOffsetIndexReference(toOffsetIndexReference(columnChunk));
+                        blockMetaData.addColumn(column);
+                    }
+                    catch (KeyAccessDeniedException e) {
+                        ColumnChunkMetaData column = new HiddenColumnChunkMetaData(columnPath, filePath);
```

Comment:
shall we add a log in case of KeyAccessDeniedException?

Reply:
Yeah,  Let me figure out how to add it. Not see log4j being used in Presto-Parquet project.

Reply:
It seems we need to add the following dependency to the subproject if we want to log in presto-parquet. I am afraid of doing that because just for adding this line of log, we need to add a new dependency. Please let me know you. thoughts

<dependency>
    <groupId>com.facebook.airlift</groupId>
    <artifactId>log</artifactId>
</dependency>